### PR TITLE
Switch to UUID for event IDs

### PR DESCRIPTION
**Pull request by MarcelMue**
# Changes
Switches the 5 character random string UIDs to RFC 4122 UUIDs.

Closes: #901

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Change the event ID representation from a 5 character random string to a UUID. 
```

### CODE REVIEW

File: pkg/template/resource.go [hunk 1]
```diff
@@ -136,13 +135,13 @@ func applyParamToResourceTemplate(param triggersv1.Param, rt json.RawMessage, ol
 	return bytes.Replace(rt, []byte(paramVariable), []byte(param.Value), -1)
 }
 
-// UID generates a random string like the Kubernetes apiserver generateName metafield postfix.
-var UID = func() string { return rand.String(5) }
+// UUID generates a Universally Unique IDentifier following RFC 4122.
+var UUID = func() string { return uuid.New().String() }
```
Comment (MarcelMue) [inline pkg/template/resource.go:139]: I have currently changed the naming everywhere from `uid` to `uuid` - I am not certain if this is really required.

`uuid` is more clear as to what it is.
`uid` is a less breaking change (and is the name k8s uses AFAIK).

Comment (MarcelMue) [reply pkg/template/resource.go:139]: cc @wlynch in case you have suggestions.

Comment (wlynch) [reply pkg/template/resource.go:139]: I agree - I'm leaning towards leaving things as `uid` based on your reasoning + so we don't have to bother with deprecation.

**Review by wlynch — COMMENTED** [review]
Thanks for picking this up!

File: pkg/template/resource.go (hunk 1 above)
Comment (MarcelMue) [reply pkg/template/resource.go:139]: This is done now - the change is not only very internal, renaming only the internal functions properly.

**Comment by MarcelMue** [comment]
I am not aware of any docs which should be changed. None that I could find mention the ID structure.

**Review by wlynch — APPROVED** [review]
/lgtm

**Pull request merged**